nvidia-bert/ort_addon/ort_supplement/ort_supplement.py

```diff
@@ -43,99 +50,94 @@ def setup_onnxruntime_with_mpi(args):
-        device, 
-        gradient_accumulation_steps=args.gradient_accumulation_steps,
-        world_rank=args.world_rank, world_size=args.world_size,
-        use_mixed_precision = True if args.fp16 else False,
-        allreduce_post_accumulation = True if args.allreduce_post_accumulation else False,
-        deepspeed_zero_stage = 1 if args.deepspeed_zero_stage else 0,
-        _opset_version = 12)
-
-    if args.fp16:
-        setattr(args, 'ort_loss_scale', LossScaler(model.loss_scale_input_name, True, up_scale_window=2000))
-
-    return model
-
-from ort_supplement.lr_schedules import SCHEDULES
-def get_lr(args, training_steps, schedule='warmup_poly'):
-    if args.max_steps == -1:
-        return args.learning_rate
-
-    schedule_fct = SCHEDULES[schedule]
-    return args.learning_rate * schedule_fct(training_steps / args.max_steps, args.warmup_proportion)
-
-def run_ort_training_step(args, global_step, training_steps, model, batch):
-    input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels = batch
-
-    lr = get_lr(args, global_step, args.schedule)
-    learning_rate = torch.tensor([lr])
-    if args.fp16:
-        loss_scale = torch.tensor([args.ort_loss_scale.loss_scale_])
-        loss = model.train_step(input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels, learning_rate, loss_scale)
-        all_finite = 1
-        if isinstance(loss, (list, tuple)):
-            assert len(loss) == 2
-            loss, all_finite = loss
-    else:
-        loss = model(input_ids, segment_ids, input_mask, masked_lm_labels, next_sentence_labels, learning_rate)
+    model_desc = {
+        'inputs': [
+            ('input_ids', [args.train_batch_size, args.max_seq_length]),
+            ('segment_ids', [args.train_batch_size, args.max_seq_length]),
+            ('input_mask', [args.train_batch_size, args.max_seq_length]),
+            ('masked_lm_labels', [args.train_batch_size, args.max_seq_length]),
+            ('next_sentence_labels', [args.train_batch_size, 2])
+        ],
+        'outputs': [
+            ('loss', [], True)
+        ]
+    }
+
+    # TRAINING OPTIMIZER SPECIFICATION
+    param_optimizer = list(model.named_parameters())
+    no_decay = ['bias', 'gamma', 'beta', 'LayerNorm']
+    optim_config = optim.LambConfig(
+        lr=args.learning_rate, alpha=0.9, beta=0.999, lambda_coef=0.01, epsilon=1e-6,
+        do_bias_correction=True,
+        params = [{
+            'params' : [n for n, p in param_optimizer if any(nd in n for nd in no_decay)],
+            'alpha': 0.9, 'beta': 0.999, 'lambda_coef': 0.00, 'epsilon': 1e-6
+        }]
+    )
+
+    # LEARNING RATE SCHEDULE SPECIFICATION
+    lr_scheduler = optim.lr_scheduler.LinearWarmupLRScheduler(
+        total_steps=int(args.max_steps), warmup=args.warmup_proportion)
+
+    # DYNAMIC LOSS SCALING SPECIFICATION (mixed precision only)
+    loss_scaler = ort_amp.loss_scaler.DynamicLossScaler()
```

Comment:
If mixed precision is enabled, `DynamicLossScaler` is used by default if none is specified, so this line is redundant

Reply:
Is there a documentation page we can refer end users to? Otherwise hiding it may make it seem like the switch doesn't exist at all.

Reply:
We added a comprehensive docstrings to the classes, so the users can do something like

```python
help(onnxruntime.training.orttrainer.ORTTrainer)
help(onnxruntime.training.orttrainer.ORTTrainerOptions)
```

and learn about it. I am working with @natke to auto generate PyTorch Frontend API and publish at onnxruntime.ai, but this is not there yet